### PR TITLE
[MINOR][DOC] Add class document for PruneFileSourcePartitions and PruneHiveTablePartitions

### PR DESCRIPTION
**Pull request by fuwhu**
### What changes were proposed in this pull request?
Add class document for PruneFileSourcePartitions and PruneHiveTablePartitions.

### Why are the changes needed?
To describe these two classes.

### Does this PR introduce any user-facing change?
no

### How was this patch tested?
no

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitions.scala [hunk 1]
```diff
@@ -26,6 +26,18 @@ import org.apache.spark.sql.catalyst.rules.Rule
 import org.apache.spark.sql.execution.datasources.v2.{DataSourceV2ScanRelation, FileScan}
 import org.apache.spark.sql.types.StructType
 
+/**
+ * Prune the partitions of file source based table using partition filters. Currently, this rule
+ * is applied to [[HadoopFsRelation]] and [[DataSourceV2ScanRelation]].
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitions.scala]: `DataSourceV2ScanRelation` -> `FileScan`

Comment (fuwhu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitions.scala]: @cloud-fan updated as : 

DataSourceV2ScanRelation -> DataSourceV2ScanRelation with FileScan
HadoopFsRelation -> HadoopFsRelation with CatalogFileIndex

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Comment by fuwhu** [comment]
thank you all for help.

**Pull request closed without merging**